universal-application-tool-0.0.1/app/services/question/ReadOnlyQuestionServiceImpl.java

```diff
@@ -17,49 +22,62 @@
 import services.question.types.QuestionType;
 import services.question.types.RepeaterQuestionDefinition;
 import services.question.types.ScalarType;
-import views.admin.GroupByKeyCollector;
 
 public final class ReadOnlyQuestionServiceImpl implements ReadOnlyQuestionService {
 
   private final ImmutableMap<Path, ScalarType> scalars;
   private final ImmutableMap<Long, QuestionDefinition> questionsById;
   private final ImmutableMap<Path, QuestionDefinition> questionsByPath;
   private final ImmutableSet<QuestionDefinition> upToDateQuestions;
+  private final ActiveAndDraftQuestions activeAndDraftQuestions;
 
   private Locale preferredLocale = Locale.US;
 
-  public ReadOnlyQuestionServiceImpl(ImmutableList<QuestionDefinition> questions) {
-    checkNotNull(questions);
+  public ReadOnlyQuestionServiceImpl(Version active, Version draft) {
+    checkNotNull(active);
+    checkState(active.getLifecycleStage().equals(LifecycleStage.ACTIVE));
```

Comment:
Suggestion (blocking): provide error message in second arg to `checkState`

Reply:
Done.